frontend/src/components/input-output/configure-ds/ConfigureDs.jsx

```diff
@@ -50,11 +50,14 @@ function ConfigureDs({
   }, [formData]);
 
   useEffect(() => {
-    if (connDetails && connDetails.connector_id !== selectedSourceId) {
-      setFormData({});
-    } else {
-      setFormData(metadata);
-    }
+    const { connector_id: connectorId } = connDetails || {};
+
+    // Check if connectorId matches selectedSourceId and metadata is available
+    const shouldSetMetadata = connectorId === selectedSourceId && metadata;
+    if (!shouldSetMetadata) return;
+
+    // Set formData based on the condition
+    setFormData(shouldSetMetadata ? metadata : {});
```

Comment:
@kirtimanmishrazipstack As we're returning if the `shouldSetMetadata` is `false`, it will be safe to directly assign `metadata` directly to `setFormData`.

Reference: `setFormData(metadata)`